frontend/src/components/3d.tsx

```diff
@@ -67,3 +114,55 @@ export function animate(): void {
     renderer.render(scene, camera);
     requestAnimationFrame(animate);
 }
+
+function dispatchBuildingEvent(eventData?: BuildingEventData): void {
+    console.log('>> dispatchBuildingEvent', eventData);
+    const event = new CustomEvent<BuildingEventData>('building-event', {
+        detail: eventData
+    });
+    document.body.dispatchEvent(event);
+}
+
+function onMouse(event: MouseEvent): void {
+    // calculate mouse position in normalized device coordinates
+    // (-1 to +1) for both components.
+    // The X coord needs to be adjusted to take sidepanel into account
+    mouse.x =
+        ((event.clientX - container.offsetLeft) / container.clientWidth) * 2 -
+        1;
+    mouse.y = -(event.clientY / container.clientHeight) * 2 + 1;
+
+    raycaster.setFromCamera(mouse, camera);
+
+    // calculate objects intersecting the picking ray
+    intersections = raycaster.intersectObjects(plywoodSheets, true);
+
+    const mouseMsg: String = `event.x:${event.x} event.y:${event.y} x:${mouse.x} y:${mouse.y}`;
+    if (intersections.length > 0) {
+        const plywoodId: String = intersections[0].object.userData.id;
+        if (plywoodId) {
+            console.log('clicked plywood sheet id: ' + plywoodId);
+            console.log(
+                `${mouseMsg} And another one HITS THE TARGET!`,
+                intersections
+            );
+            const eventData: BuildingEventData = {
+                type: 'building-clicked',
+                data: {
+                    type: 'plywood',
+                    id: plywoodId
+                }
+            };
+            dispatchBuildingEvent(eventData);
+        }
+    } else {
+        console.log(mouseMsg);
+    }
+}
+
+export function registerMouseEvent(): void {
+    console.log('registering 3D-model mouse event listener');
+    //window.addEventListener( 'mousemove', onMouse, false );
```

Comment:
Consider removing commented code

Reply:
This was left purely out of convenience since it's been used a lot.
But yeah, I know :)

Reply:
Removed